### PR TITLE
Add Pay as Business option for invoices sent to an individual who is admin of their primary workspace

### PR DESCRIPTION
**Pull request by VickyStash**
### Details

This PR implements the Pay as business option in the App for the individuals who are admins of their primary workspace.
Previous [PR](https://github.com/Expensify/App/pull/43036) with this functionality was reverted.

### Fixed Issues

$ https://github.com/Expensify/App/issues/40438
PROPOSAL: N/A

### Tests

- [x] Verify that no errors appear in the JS console

1. Login to the app as User A. The user should be the admin of any workspace.
2. Go to FAB -> Select "Send invoice"
3. Send Invoice to the User B (user should be the admin of his primary workspace)
5. Login as User B and open the invoice room from step 3.
6. Tap Pay button and select `Pay as business` -> `Pay elsewhere`
7. Verify that:
- the invoice got paid
-  the room is turned into B2B invoices room, the title and avatars got updated.
8. Login as User A and open the same invoice room. Verify B2B invoices room looks as expected for the invoice sender as well.

### Offline tests

Repeat steps 5-7 from the **Tests** section.

### QA Steps

- [x] Verify that no errors appear in the JS console

Same as in the **Tests** section

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

android_sender

https://github.com/Expensify/App/assets/23176449/c17c9e34-ca85-4602-8041-68cee1ab197a

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/23176449/bf753af6-0113-4643-b033-537ad5e3cdbd

android_web_receiver1

android_web_sender

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/23176449/ef317379-1a1f-430f-96c4-0c2fa3695b60

ios_sender

</details>

<details>
<summary>iOS: mWeb Safari</summary>

ios_web_sender

https://github.com/Expensify/App/assets/23176449/c37be27f-f410-4316-af60-4e31a94b7850

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

web_sender

https://github.com/Expensify/App/assets/23176449/afad9566-916d-4428-9aee-e3a61148171c

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/23176449/cad3000f-e1d8-43bf-afc9-dc10e84ab851

desktop_sender

</details>

### CODE REVIEW

**Comment by VickyStash** [comment]
ON HOLD till:

-  the API related issue is resolved: https://github.com/Expensify/App/issues/44818
- UI issue is fixed: https://github.com/Expensify/App/issues/44992

**Comment by VickyStash** [comment]
During the testing, I've found the next problem:

1. User A sends an invoice to User B (User A and User B never had invoices between each other before)
2. User B pays the invoice as business.
3. API sends multiple updates to the Onyx of User A. All of the data to let User A know that the invoice got paid, the invoice room got updated to B2B as well as **the info about User B policy**.
4. The updates are sent in one order but applied in another order. As you can see policy update goes first but applied last (and even if it's send last, it's still applied last, just in case).

image

_Notes to the screenshot:
1 - multiple updates from the API received with the Pusher, policy update goes first in the array
2 - report related updates applied to onyx
3 - policy update applied to onyx_

5. **Problem:** Report-related components are re-rendered only when repor/report actions are updated. The new report's data relies on the policy info which needs to be inserted. But at the moment of the re-rendering, the policy hasn't been inserted yet, which affects the UI display of invoice receiver policy-related data. 

I don't think that adding re-renders to report related components when a policy is added is a good idea from the performance side. So I guess it should be possible to affect the order Onyx updates are applied.

The App calls `Onyx.update` method
https://github.com/Expensify/App/blob/f81addc7f94b18bcf70c99ae6407e857bc90245e/src/libs/actions/User.ts#L629-L631

I saw there were some updates in Onyx related to this functionality, here is [the PR](https://github.com/Expensify/react-native-onyx/pull/490). And I tested my PR over onyx version 2.0.57 and had the same result.
<details>

  <summary>Screenshot of example using onyx 2.0.57</summary>
example1

</details>

@paultsimura Since you were working on the mentioned Onyx PR related to `Onyx.update` operation, maybe you have an idea why policy update is applied last and how can I affect it?

cc @cristipaval

**Comment by VickyStash** [comment]
@paultsimura pushJSON is the array:
image

> Is it a single Onyx.update operation, or a couple of them?

Single operation with array passed.

**Comment by VickyStash** [comment]
@paultsimura Do you have any thoughts?

**Comment by VickyStash** [comment]
I've investigated [the issue above](https://github.com/Expensify/App/pull/44970#issuecomment-2236406756) more, and overall when API provides updates to the app [it calls Onyx.update](https://github.com/Expensify/App/blob/ea09013fca8e8145ab5f44e17091e46a482792fa/src/libs/actions/User.ts#L629) with an array of updates.
Inside _Onyx.update_ the updates can be grouped, but overall it's [applied using Promise.all](https://github.com/Expensify/react-native-onyx/blob/c53826bbfe31e423fad0c33b0cb9ab097b88f128/lib/Onyx.ts#L815), so I can't affect the order updates are applied.
There was an idea to firstly push a new policy, and only then the report-related updates. But as it was internally discussed with @cristipaval we decided to fix this FE issues  on the FE side, so this commit https://github.com/Expensify/App/pull/44970/commits/5123bb56f5f6ba7f8bdccaa05d2b914eb9f53a2d adds updates to fix the problem.

**Comment by jjcoffee** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by jjcoffee** [comment]
@VickyStash Can you confirm if there's anything specifically new to test for here since the previous PR?

**Comment by VickyStash** [comment]
@jjcoffee I think it's pretty the same, just pay extra attention to the test case mentioned in this [comment](https://github.com/Expensify/App/pull/44970#issuecomment-2236406756).

**Comment by jjcoffee** [comment]
@VickyStash Testing well so far! There's a very minor flicker/glitch immediately after paying where User A's header displays the workspace name (`Collect Fresh` in the video) for a split-second before displaying the correct header (User B's workspace name). User A is to the right in the video below:

https://github.com/user-attachments/assets/77f9ba43-93f5-4562-830e-6bd3ed9f241d

**Comment by jjcoffee** [comment]
Just testing https://github.com/Expensify/App/issues/44818 (which led to the previous PR being reverted) to make sure it's all fixed. 

I'm not 100% sure on the expected result when a workspace is deleted but it seems somewhat unexpected that the avatar in the invoice room reverts to the default one (note that all the archived workspace chats retain the workspace avatar):

image

I also see that after User B deletes all workspaces, invoices from User A sent personally to User B don't show up for User B. Is this something we need to handle here?

**Comment by VickyStash** [comment]
> Just testing #44818 (which led to the previous PR being reverted) to make sure it's all fixed.
> 
> I'm not 100% sure on the expected result when a workspace is deleted but it seems somewhat unexpected that the avatar in the invoice room reverts to the default one (note that all the archived workspace chats retain the workspace avatar):
> 
> image
> 
> I also see that after User B deletes all workspaces, invoices from User A sent personally to User B don't show up for User B. Is this something we need to handle here?

@jjcoffee I'm not sure if API supports workspace removal for the invoice rooms. Maybe @cristipaval can confirm the expected behavior.

**Comment by cristipaval** [comment]
> > @VickyStash Testing well so far! There's a very minor flicker/glitch immediately after paying where User A's header displays the workspace name (`Collect Fresh` in the video) for a split-second before displaying the correct header (User B's workspace name). User A is to the right in the video below:
> 
> @jjcoffee Yeah, there is a ~second when the new receiver policy isn't written in the Onyx, so the app can't show its info cause it doesn't have it. I'm not sure if I can fix it without workarounds, but I can give it a try. cc @cristipaval

I wouldn't consider this a blocker of this PR. I think it can be fixed in a follow-up.

**Comment by cristipaval** [comment]
> > Just testing #44818 (which led to the previous PR being reverted) to make sure it's all fixed.
> > I'm not 100% sure on the expected result when a workspace is deleted but it seems somewhat unexpected that the avatar in the invoice room reverts to the default one (note that all the archived workspace chats retain the workspace avatar):
> > image
> > I also see that after User B deletes all workspaces, invoices from User A sent personally to User B don't show up for User B. Is this something we need to handle here?
> 
> @jjcoffee I'm not sure if API supports workspace removal for the invoice rooms. Maybe @cristipaval can confirm the expected behavior.

I need to deep dive into the backend code to check this.

**Comment by jjcoffee** [comment]
I've found some unusual behaviour whilst testing on iOS (I'm pretty sure it's unlikely to be platform specific though). This is with an invoice room existing between User A and User B, then sending a new invoice (personally) to User B.

1. Once paid the expense just seems to disappear (which is the issue [here](https://github.com/Expensify/App/pull/43036#issuecomment-2194549918) that we decided to leave for followup), but I can't find it under User B's workspace, only via a search.
2. Previous expense amount displays as a negative value
3. Previous expense displays with User A's workspace avatar missing.

(Note this was a fresh sign-in on iOS in case that helps).

https://github.com/user-attachments/assets/d7e08e23-0c3c-43a2-bc86-bc70e39ba357

**Comment by VickyStash** [comment]
@jjcoffee I've tested the scenario several times on ios+web and on the web+desktop, and there is no disappearing for me. 
See the video, maybe I miss something?

https://github.com/user-attachments/assets/1a31e6cf-8842-4e65-a683-3bf01ea9b71a

**Comment by VickyStash** [comment]
@jjcoffee I've tried to logout/login on web, and then on ios and both times it was okay

https://github.com/user-attachments/assets/1d73dece-97c6-4b89-a826-47fc03c63ac9

https://github.com/user-attachments/assets/15ca6420-3c06-4cbc-8838-57be54bd0f9b

**Comment by VickyStash** [comment]
@jjcoffee Could you please check if you still can reproduce that issue?

**Comment by jjcoffee** [comment]
@VickyStash Hmm yeah I can't repro either now, maybe it was a temporary BE glitch!

Should the invoice chat be visible under User B's primary workspace if I switch to it via the workspace switcher? I can only access it from the main view ("Expensify") currently, whereas for User A the invoice chat appears under their workspace.

**Comment by VickyStash** [comment]
> Should the invoice chat be visible under User B's primary workspace if I switch to it via the workspace switcher? I can only access it from the main view ("Expensify") currently, whereas for User A the invoice chat appears under their workspace.

@cristipaval @davidcardoza 

1) Should the invoice room chat appear under the workspace chats view for the invoice sender?
2) Should the invoice room chat appear under the workspace chats view for the invoice payer?

**Comment by cristipaval** [comment]
> > Should the invoice chat be visible under User B's primary workspace if I switch to it via the workspace switcher? I can only access it from the main view ("Expensify") currently, whereas for User A the invoice chat appears under their workspace.
> 
> @cristipaval @davidcardoza
> 
> 1. Should the invoice room chat appear under the workspace chats view for the invoice sender?
> 2. Should the invoice room chat appear under the workspace chats view for the invoice payer?

Great question! I'll discuss this internally

**Comment by davidcardoza** [comment]
>Should the invoice room chat appear under the workspace chats view for the invoice sender?
Should the invoice room chat appear under the workspace chats view for the invoice payer?

Discussed here - https://expensify.slack.com/archives/CSL3XBCCR/p1722944405023039

**Comment by jjcoffee** [comment]
All working well now from my testing! We've agreed the Onyx [flicker](https://github.com/Expensify/App/pull/44970#issuecomment-2265431415) should be handled as a follow-up, but I'm not sure about the avatar reverting to the default one when [deleting all workspaces](https://github.com/Expensify/App/pull/44970#issuecomment-2265445754) - should this be a blocker @cristipaval?

**Comment by cristipaval** [comment]
> but I'm not sure about the avatar reverting to the default one when [deleting all workspaces](https://github.com/Expensify/App/pull/44970#issuecomment-2265445754) - should this be a blocker @cristipaval?

I think this is also a NAB

**Review by jjcoffee — APPROVED** [review]
LGTM!

**Comment by cristipaval** [comment]
Thanks a lot for the rigorous testing @jjcoffee! 🙌 
Thanks a million to you as well, @VickyStash, for pushing this.

**Review by cristipaval — APPROVED** [review]

File: src/pages/home/HeaderView.tsx [hunk 1]
```diff
@@ -129,7 +131,7 @@ function HeaderView({report, personalDetails, parentReport, parentReportAction,
 
     const shouldShowSubscript = ReportUtils.shouldReportShowSubscript(report);
     const defaultSubscriptSize = ReportUtils.isExpenseRequest(report) ? CONST.AVATAR_SIZE.SMALL_NORMAL : CONST.AVATAR_SIZE.DEFAULT;
-    const icons = ReportUtils.getIcons(reportHeaderData, personalDetails);
+    const icons = ReportUtils.getIcons(reportHeaderData, personalDetails, null, '', -1, undefined, invoiceReceiverPolicy);
```
Comment (rojiphil) [inline src/pages/home/HeaderView.tsx:134]: When we fetched icons (and also report name), we also had to pass `policy` here. Not doing this caused issue #48710

**Pull request merged**